activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -86,7 +86,7 @@ def exist?(key)
       end
     end
 
-    def url_for_direct_upload(key, expires_in:, content_type:, content_length:, checksum:)
+    def url_for_direct_upload(key, expires_in:, content_type:, content_length:, checksum:, **)
```

Comment:
I don't think Azure supports specifying [custom metadata headers via URLs](https://github.com/Azure/azure-storage-ruby/blob/05ea58af3859edda790e769be5beba7fcfb73c0f/common/lib/azure/storage/common/core/auth/shared_access_signature_generator.rb#L381), but you can still implement `headers_for_direct_upload` and `custom_metadata_headers`.